firebase-android-release-dashboard/src/hooks/useReleaseStateUpdates.js

```diff
@@ -0,0 +1,68 @@
+import {
+  collection,
+  onSnapshot,
+  orderBy,
+  query,
+} from "firebase/firestore";
+import {useEffect, useRef, useState} from "react";
+import {db} from "../firebase";
+
+/**
+ * Custom hook to get releases with state updates.
+ *
+ * This hook uses the Firestore onSnapshot listener to get updates to the
+ * releases collection. It then filters the updates to only include releases
+ * that have had their state changed.
+ *
+ * @return {Array} The releases with state updates
+ */
+function useReleaseStateUpdates() {
+  const [releasesWithStateUpdates, setReleasesWithStateUpdates] = useState([]);
+  const previousStates = useRef({});
+
+  useEffect(() => {
+    const releasesCollection = collection(db, "releases");
+    const releasesQuery = query(releasesCollection,
+        orderBy("releaseDate", "desc"));
+
+    const unsubscribe = onSnapshot(releasesQuery, (snapshot) => {
+      let updatedReleases = [];
+
+      snapshot.docChanges().forEach((change) => {
+        const data = change.doc.data();
```

Comment:
There's a decent amount going on here; such that I feel it worth breaking up into different functions.

Reply:
I split the function up in https://github.com/firebase/firebase-release-dashboard/pull/28/commits/ac982bf2043e9c19ee0b7a6d77ffc553a04d2d01